client/navigation/utils.ts

```diff
@@ -177,45 +210,65 @@ export const sortMenuItems = ( menuItems ) => {
  * @return {Object} Mapped menu items and categories.
  */
 export const getMappedItemsCategories = (
-	menuItems,
-	currentUserCan = null
+	menuItems: Array< Item >,
+	currentUserCan: Function | null = null
 ) => {
-	const categories = { ...defaultCategories };
-
-	const items = sortMenuItems( menuItems ).reduce( ( acc, item ) => {
-		// Set up the category if it doesn't yet exist.
-		if ( ! acc[ item.parent ] ) {
-			acc[ item.parent ] = {};
-			menuIds.forEach( ( menuId ) => {
-				acc[ item.parent ][ menuId ] = [];
-			} );
-		}
+	const categories: {
+		[ key: string ]: Category | object;
```

Comment:
Similar as above, what is object here? or is it `{ [ key: string ]: Array< Item > }` as used on line 223?

(sorry not sorry if this is too much information haha)
One thing about `|` when not used as a enum, is that when parsing it you might have to cast it to that type (you might of already did this somewhere). You can do that by using `(<Type> item)` or the `as` keyword -> `item as Category`.
One cool thing, which might be handy for handling two different types is the ` is ` [keyword](https://www.typescriptlang.org/docs/handbook/advanced-types.html), especially for requests where you either get an Error back or the data.
```
function isError(data: MenuItems | Error): data is Error {
  return (data as Error).error !== undefined;
}
```
If we use it in a `if` statement, as `if (isError(data))` the else state will automatically set the type to `MenuItems` not having to do any casting.